## **Two sample z test** 

We want to know if Class1's average score on 20 marks is better than Class2's. We have the information that the standard deviation for the two classes equal $2.05$. Then we collect the data of 20 students of Class1 and 20 students of Class2 by using random samples and record their marks. Finally, we also set our $\alpha$ value (significance level) to be 0.05.

In [1]:
import numpy as np
class1=np.array([13. , 14.6, 10.2, 13.4,  9.6, 16. , 13.8, 10.6, 11.8, 12.4, 14.2,
       13.4, 12.8, 15.6, 13. ,  9.8, 16. , 12. , 10.2, 14. ])
class2=np.array([12.6 , 14.4 ,  9.2, 12.6,  8.8 , 15.6, 13.4, 10.4, 10.9,
       11.6, 13.5, 13, 12.6, 15.4, 12.3 ,  9.3, 15.6, 11.3,9.8, 13. ])

We want to test the hypothesis $\mu_1=\mu_2$ vs $\mu_1>\mu_2$ (Class1 is better than Class2)

In [2]:
m1=np.mean(class1)
m2=np.mean(class2)
[m1,m2]

[12.819999999999999, 12.265000000000002]

We now calculate the z score 
$\frac{}{}$  


In [55]:
z=(m1-m2)/np.sqrt(2.05**2/20+2.05**2/20)
z

0.8561288299480188

Critical region of the test if $\alpha=0.05$

In [43]:
zcritical=norm.ppf(1-0.05)
zcritical

1.6448536269514722

We accept $H_0$, equality of means since $z\leq zcritical$

The p value method

https://www.statsmodels.org/stable/generated/statsmodels.stats.weightstats.ztest.html

In [49]:
from statsmodels.stats.weightstats import ztest as ztest
zscore,pvalue=ztest(class1, class2,value=0)
[zscore,pvalue]

[0.8553191251187071, 0.39237450259254825]

We accept the assumption of equality of means

## **Two sample t-test**

t-test in case of equality of unkown variances

Researchers want to know whether or not two different species of plants have the same mean height. To test this, they collect a simple random sample of 20 plants from each species.

Use the following steps to conduct a two sample t-test to determine if the two species of plants have the same height.

Data

In [6]:
import numpy as np

group1 = np.array([14, 15, 15, 16, 13, 8, 14, 17, 16, 14, 19, 20, 21, 15, 15, 16, 16, 13, 14, 12])
group2 = np.array([15, 17, 14, 17, 14, 8, 12, 19, 19, 14, 17, 22, 24, 16, 13, 16, 13, 18, 15, 13])

Method 1 : the critical region method

In [13]:
[m1,m2]=[np.mean(group1),np.mean(group2)]
[m1,m2]

[15.15, 15.8]

In [14]:
[s1,s2]=[np.std(group1),np.std(group2)]
[s1,s2]

[2.7798381247835278, 3.5014282800023193]

In [15]:
[sp1,sp2]=[np.std(group1,ddof=1),np.std(group2,ddof=1)]
[sp1,sp2]

[2.852053738328889, 3.592389616661135]

Method 2 : the p-value method

Performing a t-test

The two hypotheses for this particular two sample t-test are as follows:

$H_0$: $\mu_1 = \mu_2$ (the two population means are equal)

$H_A$: $\mu_1\neq \mu_2$ (the two population means are not equal)

https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.ttest_ind.html

First, we have to decide if the variances are equal or not 

Test the equality of variance? If we know that we have Gaussian data we may use a Fisher test

In [22]:
from scipy.stats import f
F = np.var(group1) / np.var(group2)
alpha = 0.05 
df1 = len(group1) - 1
df2 = len(group2) - 1
p_value = f.cdf(F, df1, df2)
p_value

0.1614538528876288

We accept the assumption of equality of variances and perform the $t$ test under this assumption

In [3]:
import scipy.stats as stats

#perform two sample t-test with equal variances
stats.ttest_ind(a=group1, b=group2, equal_var=True)

Ttest_indResult(statistic=-0.6337397070250238, pvalue=0.5300471010405257)

Because the p-value of our test (0.53005) is greater than alpha = 0.05, we fail to reject the null hypothesis of the test. We do not have sufficient evidence to say that the mean height of plants between the two populations is different.

## **Paired t-test**

Suppose we want to know whether a certain study program significantly impacts student performance on a particular exam. To test this, we have 15 students in a class take a pre-test. Then, we have each of the students participate in the study program for two weeks. Then, the students retake a test of similar difficulty.

To compare the difference between the mean scores on the first and second test, we use a paired samples t-test because for each student their first test score can be paired with their second test score.

Data

In [4]:
pre = [88, 82, 84, 93, 75, 78, 84, 87, 95, 91, 83, 89, 77, 68, 91]
post = [91, 84, 88, 90, 79, 80, 88, 90, 90, 96, 88, 89, 81, 74, 92]

Conduct a Paired Samples T-Test

the paired samples t-test uses the following null and alternative hypotheses:

$H_0$: The mean pre-test and post-test scores are equal

$H_A$:The mean pre-test and post-test scores are not equal

https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.ttest_rel.html

In [5]:
import scipy.stats as stats

#perform the paired samples t-test
stats.ttest_rel(pre, post)

Ttest_relResult(statistic=-2.9732484231168796, pvalue=0.01007144862643272)

Since the p-value (0.0101) is less than 0.05, we reject the null hypothesis. We have sufficient evidence to say that the true mean test score is different for students before and after participating in the study program.